pkg/daemon/ceph/client/upgrade.go

```diff
@@ -104,23 +102,30 @@ func GetAllCephDaemonVersions(context *clusterd.Context, clusterName string) (*C
 }
 
 // EnableMessenger2 enable the messenger 2 protocol on Nautilus clusters
-func EnableMessenger2(context *clusterd.Context) error {
-	_, err := context.Executor.ExecuteCommandWithOutput(false, "", "ceph", "mon", "enable-msgr2")
+func EnableMessenger2(context *clusterd.Context, clusterName string) error {
+	args := []string{"mon", "enable-msgr2"}
+	buf, err := NewCephCommand(context, clusterName, args).Run()
+	output := string(buf)
```

Comment:
if err is not nil, `buf` will be invalid. But even if it is nil i guess the casting should not fail. Better to cast after the check for err?

Reply:
+1 good find